File: Java/UPCMatricula/src/models/Curso.java
package models;

import java.util.ArrayList;

public class Curso {
    private String nombre;
    ArrayList<Alumno> alumnos = new ArrayList<Alumno>();

    public Curso(String nombre) {
        this.nombre = nombre;
    }
    
    public void agregarAlumno(Alumno alumno) {
        alumnos.add(alumno);
    }
    
    public String listarMatriculados() {
        String list = "";
        list += "Curso: " + this.nombre + "\n\n";
        for(Alumno alumno : alumnos) {
           list += alumno + "\n";
        }
        return list;
    }
}

File: PHP/UPCMatricula/models/Curso.php
<?php

class Curso {
    private $nombre;
    private $alumnos = [];

    public function __construct($nombre) {
        $this->nombre = $nombre;
    }

    public function agregarAlumno($alumno) {
        $this->alumnos[] = $alumno;
    }

    public function listarMatriculados() {
        $listado = "Curso: " . $this->nombre . "\n\n";
        foreach ($this->alumnos as $alumno) {
            $listado .= $alumno . "\n";
        }
        return $listado;
    }
}
